internal/dev-tools/notice.go

```diff
@@ -46,3 +49,34 @@ func GenerateNotice(overrides, rules, noticeTemplate string) error {
 		generator.NoticeOutput("NOTICE.txt"),
 	)
 }
+
+func GenerateDependencyReport(overrides, rules, dependencyReportTemplate string, isSnapshot bool) error {
+	mg.Deps(mage.InstallGoNoticeGen, mage.Deps.CheckModuleTidy)
+
+	gotool.Mod.Tidy()     //nolint:errcheck // No value in handling this error.
+	gotool.Mod.Download() //nolint:errcheck // No value in handling this error.
+
+	out, _ := gotool.ListDepsForNotice()
+	depsFile, _ := os.CreateTemp("", "depsout")
+	defer os.Remove(depsFile.Name())
+	_, _ = depsFile.Write([]byte(out))
+	depsFile.Close()
+
```

Comment:
The part until here is identical to GenerateNotice. Could we create a helper function (getDepsFile) that executes the steps 
```
mg.Deps(mage.InstallGoNoticeGen, mage.Deps.CheckModuleTidy)

gotool.Mod.Tidy()     //nolint:errcheck // No value in handling this error.
gotool.Mod.Download() //nolint:errcheck // No value in handling this error.

out, _ := gotool.ListDepsForNotice()
depsFile, _ := os.CreateTemp("", "depsout")
_, _ = depsFile.Write([]byte(out))
depsFile.Close()
return depsFile
```

Reply:
You are right, I'll do that.